feat: added Clearing RecipeBook with Right-Click

Ryvu1 opened the pull request:
Right-click recipe book search to clear #112

Ryvu1 committed:
feat: added Clearing RecipeBook with Right-Click

BlayTheNinth commented on common/src/main/java/net/blay09/mods/clienttweaks/tweak/ClearRecipeBookSearch.java:
```diff
@@ -0,0 +1,41 @@
+        }
+    }
+
+    public boolean isEnabled() {
```
Missing `@Override` annotation

BlayTheNinth commented on common/src/main/java/net/blay09/mods/clienttweaks/ClientTweaksConfigData.java:
```diff
@@ -85,6 +85,9 @@ public static class Tweaks {
         public boolean creativeBreakingSupport = true;
+
+        @Comment("This option makes right clicking the recipe book search bar clear it.")
+        public boolean clearRecipeBook = true;
```
More descriptive name: `clearRecipeBookOnRightClick`

BlayTheNinth commented on common/src/main/java/net/blay09/mods/clienttweaks/tweak/ClearRecipeBookSearch.java:
```diff
@@ -0,0 +1,41 @@
+        Balm.getEvents().onEvent(ScreenMouseEvent.Click.Pre.class, this::onRightClick);
+    }
+
+    public void onRightClick(ScreenMouseEvent event) {
```
Parameter should be of type `ScreenMouseEvent.Click.Pre`

Ryvu1 committed:
Changes:
Added @Override annotation,
More descriptive name (clearRecipeBookOnRightClick),
Parameter ScreenMouseEvent.Click.Pre

BlayTheNinth commented on common/src/main/java/net/blay09/mods/clienttweaks/tweak/ClearRecipeBookSearch.java:
```diff
@@ -0,0 +1,41 @@
+            }
+        }
+    }
+    @Override
```
```suggestion
    
    @Override
```

BlayTheNinth committed:
Update common/src/main/java/net/blay09/mods/clienttweaks/tweak/ClearRecipeBookSearch.java
